alchemy/bin/bootstrap-s3.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  CreateBucketCommand,
+  GetBucketTaggingCommand,
+  ListBucketsCommand,
+  PutBucketTaggingCommand,
+  S3Client,
+} from "@aws-sdk/client-s3";
+import { loadConfig } from "@smithy/node-config-provider";
+
+export interface BootstrapS3Options {
+  region?: string;
+  prefix?: string;
+  help?: boolean;
+}
+
+const BOOTSTRAP_TAG_KEY = "alchemy:bootstrap";
+const BOOTSTRAP_TAG_VALUE = "s3-state-store";
+
+/**
+ * Generate a random suffix for bucket names to avoid conflicts
+ */
+function generateRandomSuffix(): string {
+  return Math.random().toString(36).substring(2, 8);
+}
+
+/**
+ * Find existing bootstrap bucket by checking tags
+ */
+async function findExistingBootstrapBucket(
+  s3Client: S3Client,
+  prefix: string,
+): Promise<string | null> {
+  try {
+    const listResult = await s3Client.send(new ListBucketsCommand({}));
```

Comment:
Isn't there a way to filter by tags. Scanning the entire AWS account is unacceptable